src/main/java/org/apache/jmeter/protocol/mqtt/sampler/SubscriberSampler.java

```diff
@@ -315,7 +315,11 @@ public SampleResult sample(Entry entry) {
 
         Message receivedMessage;
         while (!interrupted && null != client.getReceivedMessages() && null != client.getReceivedMessageCounter()) {
-            receivedMessage = client.getReceivedMessages().poll();
+             try{
+				receivedMessage = client.getReceivedMessages().take();			
+			} catch (InterruptedException e){
+				throw new RuntimeException(e);
+			}
```

Comment:
This was just to ensure that the code was executing. I shall change this to a normal print stack trace statement.